### PR TITLE
Home page

### PR DESCRIPTION
**Pull request by abhishektanwar**
This pull request does not have logic functionalities associated currently. Please review for UI only

FILES TO CHECK:
- src/components/Feed/Feed.jsx
- src/components/Feed/feed.css
- src/components/Header/Header.jsx
- src/components/Header/header.css
- src/components/LeftSidebar/LeftSidebar.jsx
- src/components/LeftSidebar/left-sidebar.css
- src/components/Post/Post.jsx
- src/components/Post/post.css
- src/components/Share/Share.jsx
- src/components/Share/share.css

### CODE REVIEW

File: src/pages/home/Home.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import Feed from '../../components/Feed/Feed'
+import Header from '../../components/Header/Header'
+import LeftSidebar from '../../components/LeftSidebar/LeftSidebar'
+import RightSidebar from '../../components/RightSidebar/RightSidebar'
+import './home.css'
+const Home = () => {
+  return (
+    <div>
+      {/* home apge */}
```
Comment (shivanigangadharan) [inline src/pages/home/Home.jsx]: remove comment

File: src/index.css [hunk 2]
```diff
@@ -1,3 +1,26 @@
...
 }
+
+.secondary-button{
+  background-color: unset;
+  color: black;
+  cursor: pointer;
+}
+
+.secondary-button svg{
+  /* font-size: 2rem; */
```
Comment (shivanigangadharan) [inline src/index.css]: remove comment

File: src/components/InputField/index.jsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      {showTogglePasswordButton ? (
+        <span
+          role="button"
+          className="toggle-password-button"
+          onClick={() => setShowPassword((prev) => !prev)}
+        >
+          {showPassword ? (
+            <i className="far fa-eye" style={{ fontSize: "1.6rem" }}></i>
+          ) : (
+            <i className="far fa-eye-slash" style={{ fontSize: "1.6rem" }}></i>
```
Comment (shivanigangadharan) [inline src/components/InputField/index.jsx]: avoid inline styling

File: src/components/Header/SearchBar.jsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  }, [searchInput])
+  return (
+    <form className="search-bar-container">
+      <input
+        type="text"
+        placeholder=""
+        value={searchInput}
+        onChange={(e) => setSearchInput(e.target.value)}
+      />
+      <i className="fa fa-search" style={{cursor:'pointer'}} onClick={()=>handleSearch(searchInput)}> </i>
```
Comment (shivanigangadharan) [inline src/components/Header/SearchBar.jsx:18]: suggestion : onClick can be placed in a <span> tag, inside which you can keep the icon

File: src/components/Buttons/BadgeIconButton.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+
+const BadgeIconButton = (props) => {
+  const { badgeNumber, icon, badgeIconButtonWrapper, onClick } = props;
+  return (
+    // <div>
+      <button onClick={onClick} className={`btn ${badgeIconButtonWrapper}`}>
```
Comment (shivanigangadharan) [inline src/components/Buttons/BadgeIconButton.jsx:6]: suggestion: if possible, can give a more descriptive function name instead on onClick (like onBadgeClick)

**Review by shivanigangadharan — APPROVED** [review]
Left few suggestions, can look into it and remove comments from wherever it's not reqd. Rest everything looks good, code is pretty clean and readable. 👍🏼

File: src/components/Buttons/BadgeIconButton.jsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        <span className="badge-icon badge-container">
+          {icon}
+          {badgeNumber ? (
+            <span className="badge-status-default badge-status-number badge-button-text">
+              {badgeNumber}
+            </span>
+          ) : null}
+        </span>
+        {/* {icon} */}
```
Comment (zoef-yt) [inline src/components/Buttons/BadgeIconButton.jsx]: remove commented code.

File: src/components/Buttons/BadgeIconButton.jsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+      </button>
+    // </div>
```
Comment (zoef-yt) [inline src/components/Buttons/BadgeIconButton.jsx]: remove commented code.

File: src/components/Buttons/BadgeIconButton.jsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+
+const BadgeIconButton = (props) => {
+  const { badgeNumber, icon, badgeIconButtonWrapper, onClick } = props;
+  return (
+    // <div>
```
Comment (zoef-yt) [inline src/components/Buttons/BadgeIconButton.jsx]: remove commented code.

File: src/pages/home/Home.jsx [hunk 9]
```diff
@@ -0,0 +1,22 @@
+import Feed from '../../components/Feed/Feed'
+import Header from '../../components/Header/Header'
+import LeftSidebar from '../../components/LeftSidebar/LeftSidebar'
+import RightSidebar from '../../components/RightSidebar/RightSidebar'
```
Comment (zoef-yt) [inline src/pages/home/Home.jsx]: You can use index.js and make all imports from there.

File: src/index.css [hunk 10]
```diff
@@ -1,3 +1,26 @@
 *{
   font-family: 'Roboto', sans-serif;
 }
+
+.secondary-button{
+  background-color: unset;
+  color: black;
```
Comment (zoef-yt) [inline src/index.css]: use var colours.

File: src/components/Share/Share.jsx [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+            textarea={true}
+          />
+        </div>
+        <div className="share-bottom-section flex-row">
+          <div className="share-options flex-row flex-justify-content-space-between">
+            <Button
+              buttonStyle="secondary-button margin0 flex-align-item-center body-typo-sm flex-row"
+              icon={<PermMedia htmlColor="tomato" fontSize="medium" />}
+              buttonText="Photo or video"
+            />
```
Comment (zoef-yt) [inline src/components/Share/Share.jsx]: make a button style var and pass that to the button style as you are repeating yourself.

File: src/components/RightSidebar/right-sidebar.css [hunk 12]
```diff
@@ -0,0 +1,9 @@
+.right-sidebar-container{
+  flex: 25%;
+  height: 90vh;
+}
+
+.sidebar-wrapper{
+  /* padding: 2rem; */
```
Comment (zoef-yt) [inline src/components/RightSidebar/right-sidebar.css]: remove commented code.

File: src/components/RightSidebar/RightSidebar.jsx [hunk 13]
```diff
@@ -0,0 +1,13 @@
+import './right-sidebar.css'
+const RightSidebar = () => {
+  return (
+    <div className="right-sidebar-container">
+      <div className="right-sidebar-wrapper">
+        
+      </div>
+      right saidebar
```
Comment (zoef-yt) [inline src/components/RightSidebar/RightSidebar.jsx]: sidebar* typo.

File: src/components/Post/post.css [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+.post-bottom-section{
+  margin-top: 1.4rem;
+  justify-content: space-between;
+  align-items: center;
+}
+
+.like-counter{
+  /* margin-left: rem; */
```
Comment (zoef-yt) [inline src/components/Post/post.css]: remove commented code.

File: src/components/LeftSidebar/LeftSidebar.jsx [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+    <div className="left-sidebar-container">
+      <div className="left-sidebar-wrapper">
+        <ul className="sidebar-list padding0">
+          <li className="sidebar-list-item">
+            <Button
+              buttonStyle="secondary-button margin0 flex-align-item-center flex-row"
+              icon={<HomeOutlined fontSize="large" />}
+              buttonText="Feed"
+            />
+          </li>
```
Comment (zoef-yt) [inline src/components/LeftSidebar/LeftSidebar.jsx]: Make this into a component too and send text and icon as a prop.

File: src/components/InputField/input.css [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+.invalid-field-border:hover{
+  border-color: var(--field-required-color);
+}
+
+.toggle-password-button{
+  position: absolute;
+  top: 39px;
+  right: 22px;
```
Comment (zoef-yt) [inline src/components/InputField/input.css]: Try to use relative units.

File: src/components/Header/header.css [hunk 17]
```diff
@@ -0,0 +1,85 @@
+.nav-bar{
+  padding:unset;
+}
+
+.search-bar-container{
+  width: 40%;
+  border: 1px solid #D1D1D1;
```
Comment (zoef-yt) [inline src/components/Header/header.css]: use var for volours.

File: src/components/Header/header.css [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+  padding: 10px;
+  border-radius: 5px;
+  background-color: #F9F9F9;
```
Comment (zoef-yt) [inline src/components/Header/header.css]: use var for volours.

File: src/components/Header/SearchBar.jsx [hunk 19]
```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from "react";
+import "./header.css";
+
+const SearchBar = ({handleSearch}) => {
+  const [searchInput, setSearchInput] = useState("");
+  useEffect(() => {
+    if (searchInput===""){
+      // handleSearch("")
```
Comment (zoef-yt) [inline src/components/Header/SearchBar.jsx]: remove commented code.

File: src/components/Feed/feed.css [hunk 20]
```diff
@@ -0,0 +1,10 @@
+.feed-container{
+  flex: 50%;
+  /* background-color: green; */
```
Comment (zoef-yt) [inline src/components/Feed/feed.css]: remove commented code.

**Review by zoef-yt — CHANGES_REQUESTED** [review]

File: src/components/Buttons/BadgeIconButton.jsx (hunk 5 above)
Comment (abhishektanwar) [reply src/components/Buttons/BadgeIconButton.jsx:6]: Yes, that would make code more readable. thanks for pointing this out.

File: src/components/Header/SearchBar.jsx (hunk 4 above)
Comment (abhishektanwar) [reply src/components/Header/SearchBar.jsx:18]: Yes, that could be done . But the only intention here was to keep the DOM elements as less as possible.

File: src/pages/home/Home.jsx (hunk 1 above)
Comment (abhishektanwar) [reply src/pages/home/Home.jsx]: fixed.

File: src/index.css (hunk 2 above)
Comment (abhishektanwar) [reply src/index.css]: fixed

File: src/components/InputField/index.jsx (hunk 3 above)
Comment (abhishektanwar) [reply src/components/InputField/index.jsx]: fixed

File: src/components/Buttons/BadgeIconButton.jsx (hunk 8 above)
Comment (abhishektanwar) [reply src/components/Buttons/BadgeIconButton.jsx]: fixed

File: src/components/Buttons/BadgeIconButton.jsx (hunk 7 above)
Comment (abhishektanwar) [reply src/components/Buttons/BadgeIconButton.jsx]: fixed

File: src/components/Buttons/BadgeIconButton.jsx (hunk 6 above)
Comment (abhishektanwar) [reply src/components/Buttons/BadgeIconButton.jsx]: fixed

File: src/pages/home/Home.jsx (hunk 9 above)
Comment (abhishektanwar) [reply src/pages/home/Home.jsx]: Thanks for this suggestion, it made code more readable

File: src/index.css (hunk 10 above)
Comment (abhishektanwar) [reply src/index.css]: fixed

File: src/components/Share/Share.jsx (hunk 11 above)
Comment (abhishektanwar) [reply src/components/Share/Share.jsx]: fixed

File: src/components/RightSidebar/right-sidebar.css (hunk 12 above)
Comment (abhishektanwar) [reply src/components/RightSidebar/right-sidebar.css]: fixed

File: src/components/RightSidebar/RightSidebar.jsx (hunk 13 above)
Comment (abhishektanwar) [reply src/components/RightSidebar/RightSidebar.jsx]: fixed

File: src/components/Post/post.css (hunk 14 above)
Comment (abhishektanwar) [reply src/components/Post/post.css]: fixed

File: src/components/LeftSidebar/LeftSidebar.jsx (hunk 15 above)
Comment (abhishektanwar) [reply src/components/LeftSidebar/LeftSidebar.jsx]: fixed

File: src/components/InputField/input.css (hunk 16 above)
Comment (abhishektanwar) [reply src/components/InputField/input.css]: fixed

File: src/components/Header/header.css (hunk 18 above)
Comment (abhishektanwar) [reply src/components/Header/header.css]: fixed

File: src/components/Feed/feed.css (hunk 20 above)
Comment (abhishektanwar) [reply src/components/Feed/feed.css]: fixed

File: src/components/Header/SearchBar.jsx (hunk 19 above)
Comment (abhishektanwar) [reply src/components/Header/SearchBar.jsx]: removed

File: src/components/Header/header.css (hunk 17 above)
Comment (abhishektanwar) [reply src/components/Header/header.css]: fixed

**Review by zoef-yt — APPROVED** [review]

**Pull request merged**